src/kv/untyped_map.h

```diff
@@ -225,7 +223,7 @@ namespace kv::untyped
             if (search.has_value())
             {
               changes = true;
-              state = state.put(it->first, VersionV{-v, v_, {}});
+              state = state.remove(it->first);
```

Comment:
I am concerned that removing the deletion from the state results in it being gone from the roll, and affects conflict resolution. What am I missing?

Reply:
Ok, so mostly for myself, this seems to be fine, because a read dependency on a key missing is not inherently versioned. So long as the key is not currently there, there is no conflict, even if the key was added and removed since.